From ab21f9f9ff572b1b52d9794e9794939e56c559c1 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Sun, 11 May 2014 22:32:14 -0400
Subject: [PATCH] Don't wait forever in curl_multi_wait

This seems to fix problems with some FTP servers. Odd.
---
 src/update.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/update.c b/src/update.c
index dedb56c..22f2444 100644
--- a/src/update.c
+++ b/src/update.c
@@ -567,7 +567,7 @@ static void download_wait_loop(CURLM *multi) {
       break;
     }
 
-    rc = curl_multi_wait(multi, NULL, 0, -1, NULL);
+    rc = curl_multi_wait(multi, NULL, 0, 1000, NULL);
     if (rc != CURLM_OK) {
       fprintf(stderr, "error: curl_multi_wait failed (%d)\n", rc);
       break;
-- 
1.9.2

